LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/ConcentratorDeduplication.cs

```diff
@@ -30,96 +39,110 @@ internal sealed class ConcentratorDeduplication<T> :
 
         public ConcentratorDeduplication(
             IMemoryCache cache,
-            WebSocketWriterRegistry<StationEui, string> socketRegistry,
-            ILogger<IConcentratorDeduplication<T>> logger)
+            ILogger<IConcentratorDeduplication> logger)
         {
             this.cache = cache;
-            this.socketRegistry = socketRegistry;
             this.logger = logger;
         }
 
-        public bool ShouldDrop(T frame, StationEui stationEui)
+        public async Task<Result> CheckDuplicateAsync(LoRaRequest loRaRequest, LoRaDevice? loRaDevice)
```

Comment:
Why did you make the method async? Was this to add the semaphore? You don't need that here. You can simply add an object as a field for example

`private static readonly object cacheLock = new object();`

since the concentrator deduplicator is also a singleton, in theory you could also use a non-static field, however, that would ensure, if the registration changes, you still lock the cache (which is being added as a singleton as well).

Reply:
yes it was for that. Changed it with your suggestion in ceeab95c5ff2b16fbb7bc0705afe69c9ef4544a8